### ISSUE TITLE
Sync Unity meta data with dbt model.yml metadata on run

### ISSUE THREAD

**Issue by EtherJfox** [databricks/dbt-databricks#247]
### Describe the feature
dbt tracks table meta data (name, description, column name, column description) in the model.yml. Unity also has these meta data labels. A dbt run should sync the metadata to Unity.

### Describe alternatives you've considered
Writing my own python notebook to parse manifest.json for these meta data items and iterating over it myself to add the metadata to Unity.

### Additional context
Both the create and alter statement have the ability to add table comments at table or column level.

### Who will this benefit?
Everyone using model documentation. We're creating dbt documentation as we go, Unity should benefit.

### Are you interested in contributing this feature?
I have my own method of syncing the data but I don't have any time in the foreseeable future to look into how to get this working inside the dbt-databricks framework.

**Comment by ueshin** [comment]
Hi @EtherJfox,

Could you elaborate what kind of metadata you mean?
Are they different from what `persist_docs` does?

**Comment by EtherJfox** [comment]
Didn't know the feature existed. Looks like I'm getting the table level comment on some tables, but not others, and no column comment.

Sample config

version: 2

```
models:
  - name: finance_datamart__dim_date_accounting
    description: A date dimension used in the Finance data model representing accounting periods
    columns:
      - name: date_key
        description: An integer representation of the date
    config:
      enabled: true
      file_format: delta
      +persist_docs:
        relation: true
        columns: true  
```

**Comment by ueshin** [comment]
Do we need `+` before `persist_docs` in the model property file?
According to the [General configurations](https://docs.getdbt.com/reference/model-configs#general-configurations), it's necessary for `Project file`, but not for `Property file`?

What if moving the config to the project file?

**Comment by grindheim** [comment]
@ueshin I feel like the `persist_docs` functionality could be significantly improved for the dbt-databricks implementation.

The way it currently works is by first creating the table, then sending one `alter table ... change column ... comment` after another for every single table. This takes an unnecessary long time to complete. E.g.
![image](https://user-images.githubusercontent.com/19465110/229510831-2ed34dba-b804-4723-97e3-2b3d1800fb5f.png)

However, there are two alternative approaches here that should be vastly faster.

**1. Add comments directly in `create (or replace)` statements**
Databricks supports the following syntax, which takes as long time to complete as any regular `create (or replace)` statement
```
create or replace table kumulus_dev.sindre__std.test
(id INT comment 'description', name STRING comment 'description', age INT comment 'description')
COMMENT 'this is a comment'
```
**2. Utilize the Unity Catalog API**
With unity catalog, using the unity catalog api could also be a fast option. 
https://api-docs.databricks.com/rest/latest/unity-catalog-api-specification.html

**In summary**
These are only meta operations and should be possible to update in bulk rather than comment by comment. The way it is currently, it's simply not feasible to use `persist_docs`, but it shouldn't have to be this way.

**Comment by ueshin** [comment]
cc @andrefurlan-db

**Comment by jordandakota** [comment]
> Could we open a new issue for this performance issue with syncing metadata? I think the initial topic has been resolved with persist_docs.

https://github.com/databricks/dbt-databricks/issues/507
